File: 백준/Bronze/10818. 최소， 최대/최소， 최대.swift
/// [func] getResultWithoutMethod
/// 
/// # Notes
/// - min, max 함수 없이 최소값, 최대값을 산출해내는 함수
/// 
/// # Parameters
/// - Array<Int> targetArray : 값을 산출하고자 하는 정수 배열
/// 
/// # Returns
/// - Array<Int> result : [minOfArray, maxOfArray]
/// 
func getResultWithoutMethod(targetArray: Array<Int>) -> Array<Int> {
    // [Int] minOfArray : 배열의 최소값을 저장하는 변수 (입력의 상한선으로 초기화)
    var minOfArray: Int = 1000000
    // [Int] maxOfArray : 배열의 최대값을 저장하는 변수 (입력의 하한선으로 초기화)
    var maxOfArray: Int = -1000000
    
    for i in 0..<targetArray.count {
        if (targetArray[i] < minOfArray) {
            minOfArray = targetArray[i]
        }
        if (targetArray[i] > maxOfArray) {
            maxOfArray = targetArray[i]
        }
    }
  
    return [minOfArray, maxOfArray]
}

/// [func] getResultWithMethod
/// 
/// # Notes
/// - min, max 함수를 사용하여 최소값, 최대값을 산출해내는 함수
/// 
/// # Parameters
/// - Array<Int> targetArray : 값을 산출하고자 하는 정수 배열
/// 
/// # Returns
/// - Array<Int> result : [minOfArray, maxOfArray]
/// 
func getResultWithMethod(targetArray: Array<Int>) -> Array<Int> {
    // [Int] minOfArray : 배열의 최소값을 저장하는 변수
    var minOfArray: Int = (targetArray.min() ?? 0)
    // [Int] maxOfArray : 배열의 최대값을 저장하는 변수
    var maxOfArray: Int = (targetArray.max() ?? 0)
  
    return [minOfArray, maxOfArray]
}


// [Int] N : 정수의 개수 (1 <= N <= 1,000,000)
let N: Int = Int(readLine() ?? "1") ?? 1

// [Array<Int>] arr : Int를 저장하는 배열 (-1,000,000 <= int <= 1,000,000)
let arr: Array<Int> = (readLine() ?? "0").split(separator: " ").map{Int($0) ?? 0}

let resultArr = getResultWithMethod(targetArray: arr)

print(resultArr[0], resultArr[1])

File: 백준/Bronze/10818. 최소， 최대/최소， 최대.py
# [function] getResultWithoutMethod
#
# Notes
# - min, max 메서드 없이 int list의 최소/최대값을 산출하는 함수
#
# Parameters
# - [list] arr : 값을 산출하고자 하는 list
#
# Returns
# - [tuple] (minOfArr, maxOfArr) : 최소값, 최대값 튜플
# 
def getResultWithoutMethod(arr):
    # [int] minOfArr : 최소값을 저장하는 변수 (입력 제한의 상한 기준)
    # [int] maxOfArr : 최대값을 저장하는 변수 (입력 제한의 하한 기준)
    minOfArr = 1000000
    maxOfArr = -1000000

    # list를 탐색하며 최솟값과 최댓값을 수정함
    for number in arr:
        if (number < minOfArr) :
            minOfArr = number
        if (number > maxOfArr) :
            maxOfArr = number
            
    return minOfArr, maxOfArr

# [int] N : 정수의 개수 (1 <= N <= 1,000,000)
N = int(input())

# [list] arr : 입력 받은 정수를 저장한 리스트
arr = map(int, input().split())

minOfArr, maxOfArr = getResultWithoutMethod(arr)
 
# 각 값을 출력
print(minOfArr, maxOfArr)